keps/0019-package-api-versioning.md

```diff
@@ -2,64 +2,374 @@
 kep-number: 19
 title: Versioning of Operator Packages
 authors:
+  - "@mpereira"
   - "@nfnt"
 owners:
   - "@nfnt"
 creation-date: 2019-11-05
-last-updated: 2019-11-05
+last-updated: 2019-11-21
 status: provisional
 ---
 
 # Versioning of Operator Packages
 
-## Table of Contents
+**Table of Contents**
 
-* [Summary](#summary)
-* [Motivation](#motivation)
-  * [Goals](#goals)
-  * [Non-Goals](#non-goals)
-* [Proposal](#proposal)
-  * [Support multiple API versions in the package repository](#support-multiple-api-versions-in-the-package-repository)
-  * [Install a specific application version](#install-a-specific-application-version)
-  * [Naming of package tarballs](#naming-of-package-tarballs)
-  * [Semantic versioning](#semantic-versioning)
-  * [Risks and Mitigations](#risks-and-mitigations)
-* [Implementation History](#implementation-history)
+- [Concepts](#concepts)
+    - [KUDO version](#kudo-version)
+    - [KUDO API version](#kudo-api-version)
+    - [Operator](#operator)
+    - [Operator user (user)](#operator-user-user)
+    - [Operator developer (developer)](#operator-developer-developer)
+    - [Package](#package)
+    - [Package version](#package-version)
+    - [Package Registry](#package-registry)
+    - [Application](#application)
+    - [Application version (app version)](#application-version-app-version)
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+        - [Non-functional](#non-functional)
+        - [Functional](#functional)
+    - [Non-Goals](#non-goals)
+- [Current State](#current-state)
+- [Proposals](#proposals)
+    - [With regards to package version and operator version](#with-regards-to-package-version-and-operator-version)
+        - [Singular package/operator version (largely, the current state of things)](#singular-packageoperator-version-largely-the-current-state-of-things)
+        - [Composite package/operator version](#composite-packageoperator-version)
+    - [Support multiple KUDO API versions in the package repository](#support-multiple-kudo-api-versions-in-the-package-repository)
+    - [Install a specific application version](#install-a-specific-application-version)
+    - [Naming of package tarballs](#naming-of-package-tarballs)
+    - [Semantic versioning](#semantic-versioning)
+    - [Risks and Mitigations](#risks-and-mitigations)
+- [Open Questions](#open-questions)
+- [Implementation History](#implementation-history)
+
+## Concepts
+
+### KUDO version
+
+A released version of KUDO (CLI, controller-manager), e.g. 0.7.0, 0.8.0, etc.
+
+### KUDO API version
+
+The internal version for the KUDO implementation, e.g. v1alpha1, v1beta1, v1, etc.
```

Comment:
maybe `internal API format version for the YAML files` or something along the lines?

Reply:
I like `KUDO API version`

Reply:
Yup, that is what the headline states, but `The internal version for the KUDO implementation` was a bit confusing to me what it actually means so I was trying to make it better :D